logsmanagement/flb_plugin.c

```diff
@@ -0,0 +1,982 @@
+            tmp_item_off += docker_ev_type_size;
+            memcpy(&buff->in->data[tmp_item_off++], " ", 1);
+        }
+
+        if(likely(docker_ev_action)){
+            memcpy(&buff->in->data[tmp_item_off], docker_ev_action, docker_ev_action_size);
+            tmp_item_off += docker_ev_action_size;
+            memcpy(&buff->in->data[tmp_item_off++], " ", 1);
+        }
+
+        if(likely(docker_ev_id)){
+            memcpy(&buff->in->data[tmp_item_off], docker_ev_id, docker_ev_id_size);
+            tmp_item_off += docker_ev_id_size;
+            memcpy(&buff->in->data[tmp_item_off++], " ", 1);
+        }
+
+        if(likely(docker_ev_attr.size)){
+            memcpy(&buff->in->data[tmp_item_off++], "(", 1);
+            for(int i = 0; i < docker_ev_attr.size; i++){
+                memcpy(&buff->in->data[tmp_item_off], docker_ev_attr.key[i], docker_ev_attr.key_size[i]);
+                tmp_item_off += docker_ev_attr.key_size[i];
+                memcpy(&buff->in->data[tmp_item_off++], "=", 1);
+                memcpy(&buff->in->data[tmp_item_off], docker_ev_attr.val[i], docker_ev_attr.val_size[i]);
+                tmp_item_off += docker_ev_attr.val_size[i];
+                memcpy(&buff->in->data[tmp_item_off++], ",", 1);
+                memcpy(&buff->in->data[tmp_item_off++], " ", 1);
+            }
+            tmp_item_off -= 2; // overwrite last ',' and ' ' characters with a ')' character
+            memcpy(&buff->in->data[tmp_item_off++], ")", 1);
+        }
+
+
+        memcpy(&buff->in->data[tmp_item_off++], "\n", 1);
+        m_assert(tmp_item_off == new_tmp_text_size, "tmp_item_off should be == new_tmp_text_size");
+        buff->in->text_size = new_tmp_text_size;
+    }
+    /* FLB_DOCKER_EV case end */
+
+    /* Following code is equivalent to msgpack_unpacked_destroy(&result) due 
+     * to that function call being unavailable when using dl_open() */
+    if(result.zone != NULL) {
+        dl_msgpack_zone_free(result.zone);
+        result.zone = NULL;
+        memset(&result.data, 0, sizeof(msgpack_object));
+    }
+
+    uv_mutex_unlock(&p_file_info->flb_tmp_buff_mut);
+    
+    flb_lib_free(record);
+    // FLB_OUTPUT_RETURN(FLB_OK); // Watch out! This breaks output - won't flush all pending logs
+    return 0;
+    
+}
+
+/**
+ * @brief Add a Fluent-Bit input that outputs to the "lib" Fluent-Bit plugin.
+ * @param[in] p_file_info Pointer to the log source struct where the input will
+ * be registered to.
+ * @return 0 on success, a negative number for any errors.
+ */
+int flb_add_input(struct File_info *const p_file_info){
+
+    enum return_values {
+        SUCCESS = 0,
+        INVALID_LOG_TYPE = -1,
+        FLB_PARSER_CREATE_ERROR = -2,
+        FLB_INPUT_ERROR = -3,
+        FLB_INPUT_SET_ERROR = -4,
+        FLB_OUTPUT_ERROR = -5,
+        FLB_OUTPUT_SET_ERROR = -6,
+        DEFAULT_ERROR = -7
+    };
+
+    static unsigned tag = 0; // incremental tag id to link flb inputs to outputs
```

Comment:
Neither `tag` nor `tag_s` was modified in the sequence, so we can simplify this code and discard `tag`.

Reply:
It's a `static` variable that gets modified 2 lines below (`tag++`). It is used to ensure every new Fluent-Bit input is unique.